### ISSUE TITLE
Improve `ProductTitle` Accessibility

### ISSUE THREAD

**Issue by hahnwhitney** [vtex/faststore#1935]
**Background**
I've been working on a Jest/RTL accessibility test suite for FastStore's UI components. This issue isn't a bug in the sense that the component is "broken" -- it just doesn't meet the WCAG standards.

**Describe the bug**
The [`<p>` tag on line 60](https://github.com/vtex/faststore/blob/2847a365cb0da48e493ecdb0c9f2975e672628d4/packages/components/src/molecules/ProductTitle/ProductTitle.tsx#L60) of the `ProductTitle` component currently contains both a `<div>` and a `<ul>` as children/descendants; this is not semantically correct.

```
        {(refNumber || ratingValue) && (
          <p data-fs-product-title-addendum>
            {ratingValue && <Rating value={ratingValue} />}
            {refNumber && (
              <>
                {refTag} {refNumber}
              </>
            )}
          </p>
        )}
```

**Possible Solution**
The `<p>` tag on line 60 needs to be changed to a `<div>`.

**Additional context**
To see the error, you can check out my [tests PR branch in the demoHearst repo](https://github.com/vtex-sites/demohearst.store/pull/27) and run the following command in the console:
`yarn test-jest src/components/molecules/producttitle`

![Screenshot 2023-08-07 at 10 02 54 AM](https://github.com/vtex/faststore/assets/88338049/9adc3110-33fc-428d-b140-54c71f1a2bcd)

If you'd like to print the component's DOM tree in the console, you can pull down the `RTL-test-examples` branch, and then use the `screen.debug()` command in the `ProductTitle` test:
![Screenshot 2023-08-07 at 10 06 14 AM](https://github.com/vtex/faststore/assets/88338049/af6703dc-e925-4862-83d3-86f06f89203b)

The DOM looks like this (note that I cut on the bottom of the DOM tree because the screenshot was so long):
![Screenshot 2023-08-07 at 10 08 09 AM](https://github.com/vtex/faststore/assets/88338049/5a81e88c-b05b-4813-97b0-18db35ffe65f)

**Comment by renatamottam** [comment]
Hi @hahnwhitney, thanks for reporting this issue in such detail, it's really helpful ✨
We created an internal task and will fix it soon.

**Comment by hellofanny** [comment]
Done! 💯
